src/services/schema-inference.ts

```diff
@@ -0,0 +1,26 @@
+import { client } from 'services/client';
+import { GatewayAuthTokenResponse, Schema } from 'types';
+
+export interface InferSchemaResponse {
+    documents_read: number;
+    exceeded_deadline: boolean;
+    schema: Schema;
+}
+
+const getInferredSchema = (
+    gatewayConfig: GatewayAuthTokenResponse,
+    collection: string
+): Promise<InferSchemaResponse> => {
+    const { gateway_url, token } = gatewayConfig;
+
+    const headers: HeadersInit = {};
+
+    headers.Authorization = `Bearer ${token}`;
+    headers['Content-Type'] = 'application/json';
```

Comment:
No change needed for this pr

This approach for API calls should really be handle in a single function given it is getting duplicated.